compiler/hash-lexer/src/lib.rs

```diff
@@ -306,7 +305,7 @@ impl<'a> Lexer<'a> {
             return None;
         }
 
-        let location = Span::new(offset, self.len_consumed());
+        let location = ByteRange::new(offset, self.len_consumed());
```

Comment:
This counts characters no?

Reply:
Bytes:
```rs
    fn len_consumed(&self) -> usize {
        self.offset.get()
    }
```

Reply:
However, it shouldn't ever be a value that is invalid to index into (in terms of bytes). So you shouldn't be able to end up in the middle of a unicode codepoint.

Reply:
Okay I see. Then this is fine